File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/models/GenesisError.java
package com.emerchantpay.gateway.genesisandroid.api.models;

public class GenesisError {

    private Integer code;
    private String technicalMessage;
    private String message;

    public GenesisError() {
        super();
    }

    public GenesisError(Integer code, String message, String technicalMessage) {
        super();

        this.code = code;
        this.message = message;
        this.technicalMessage = technicalMessage;
    }

    public GenesisError(Integer code, String technicalMessage) {
        super();

        this.code = code;
        this.technicalMessage = technicalMessage;
    }

    public GenesisError(String message, String technicalMessage) {
        super();

        this.message = message;
        this.technicalMessage = technicalMessage;
    }

    public GenesisError(String message) {
        super();

        this.message = message;
    }

    public Integer getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }

    public String getTechnicalMessage() {
        return technicalMessage;
    }
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/models/GenesisError.kt
package com.emerchantpay.gateway.genesisandroid.api.models

open class GenesisError {

    var code: Int? = null
        private set
    var technicalMessage: String? = null
        private set
    var message: String? = null
        private set

    constructor() : super() {}

    constructor(code: Int?, message: String, technicalMessage: String) : super() {

        this.code = code
        this.message = message
        this.technicalMessage = technicalMessage
    }

    constructor(code: Int?, technicalMessage: String) : super() {

        this.code = code
        this.technicalMessage = technicalMessage
    }

    constructor(message: String, technicalMessage: String) : super() {

        this.message = message
        this.technicalMessage = technicalMessage
    }

    constructor(message: String) : super() {

        this.message = message
    }
}
